### PR TITLE
Fix camera transforms in VR

### LINKED ISSUE #13236: three.js VR renderer does frustum culling based on only the left eye, causing premature clipping on the right eye.
##### Description of the problem

The VR renderer does frustum culling based on the frustum of the left eye camera. It applies the same culling to the right eye. As a result, when an object leaves the left eye's view, the user can see the object flash out on the corner of the right eye periphery.

Saw  https://github.com/w3c/webvr/issues/203 was linked.

https://github.com/mrdoob/three.js/pull/11523

##### Three.js version

- [x] Dev
- [ ] r89
- [ ] ...

##### Browser

- [ ] All of them
- [ ] Chrome
- [x] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

##### Hardware Requirements (graphics card, VR Device, ...)

VR Device

### PR DESCRIPTION
**Pull request by brianpeiris**
Fixes #13236

### CODE REVIEW

File: src/renderers/webvr/WebVRManager.js [hunk 1]
```diff
@@ -104,40 +108,49 @@ function WebVRManager( renderer ) {
...
 
-			standingMatrix.fromArray( stageParameters.sittingToStandingTransform );
+			tempQuaternion.setFromRotationMatrix(standingMatrix);
+			tempPosition.fromArray( pose.position );
+			tempPosition.applyQuaternion(tempQuaternion);
+			poseObject.position.add( tempPosition );
 
 		} else {
 
-			standingMatrix.makeTranslation( 0, scope.userHeight, 0 );
```
Comment (dmarcos) [inline src/renderers/webvr/WebVRManager.js]: Not using userHeight anymore is intentional?

Comment (brianpeiris) [reply src/renderers/webvr/WebVRManager.js]: The diff is misleading. This `if` block was moved up to line 113

Comment (dmarcos) [reply src/renderers/webvr/WebVRManager.js]: I see the standingMatrix and scope.userHeight is accounted for above. If position is null I think this else branch would reset the height back to 0 overriding the scope.userHeight, doesn’t it?

Comment (brianpeiris) [reply src/renderers/webvr/WebVRManager.js]: Ah, you're right. Good catch.

Comment (dmarcos) [reply src/renderers/webvr/WebVRManager.js]: Removing the else would do I think

**Comment by brianpeiris** [comment]
This also fixes the transform of objects attached to the camera. E.g. the crosshair in the webvr_cubes example.

**Comment by dmarcos** [comment]
I just tested on Vive and I cannot reproduce the culling issues. @brianpeiris 🏅

**Comment by mrdoob** [comment]
Thanks!

**Comment by RSpace** [comment]
I just tested this fix with own A-Frame / Three.js where I started to see clipping issues for the right eye on Oculus Rift after updating to A-Frame 0.8.1 / Three.js r90.

When using latest version from mrdoob#dev where this has merged in, I unfortunately still have exactly the same clipping issue for the right eye as before. So this PR hasn't fixed the issue for me, but I unfortunately lack the detailed knowledge of how transforms and culling works to fix it myself.

Anything I can do to help to fully resolve this issue?

**Comment by brianpeiris** [comment]
@RSpace Do you have a link to repro?

**Comment by dmarcos** [comment]
@brianpeiris Your PR solved the problem of objects being incorrectly culled due to the bad construction of the camera matrix. The issue described in https://github.com/mrdoob/three.js/issues/13236 has been always present since the VR logic switched to use just the left eye frustum for culling. Any example should reproduce the problem where objects are incorrectly culled on the right side of the right eye field of view. More info on https://github.com/aframevr/aframe/issues/3488

**Comment by brianpeiris** [comment]
Ah, thanks for clearing that up.

**Pull request merged**